### ISSUE TITLE
Release v0.4.23

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-ipfs#6837]
# go-ipfs 0.4.23 Release

PR: https://github.com/ipfs/go-ipfs/pull/6836
Download: https://dist.ipfs.io/go-ipfs/v0.4.23-rc2 (RC2)

Unfortunately, [go-ipfs 0.5.0](https://github.com/ipfs/go-ipfs/issues/6776) is taking longer than expected. Given the large number of fixes merged since 0.4.22, we've decided to cut another patch release.

This release contains critical fixes. Please upgrade ASAP. Importantly, we're strongly considering switching to TLS by default in go-ipfs 0.5.0 and dropping SECIO support. However, our the current TLS transport in go-ipfs 0.4.22 has a bug that can cause connections to spontaneously disconnect during the handshake.

This release fixes that bug, among many other issues. Users that _don't_ upgrade may experience connectivity issues when the network upgrades to go-ipfs 0.5.0.

## 🗺 What's left for release

* [x] Get the tests passing.
* [ ] Release Notes.

# 🚢 Estimated shipping date

January 27th, at the latest (likely earlier).

## 🔦 Highlights

* Fixes build on go 1.13
* Fixes an issue where we may not connect to providers in bitswap.
* Fixes an issue on our TLS transport where we may abort a handshake unintentionally.
* Fixes a common panic in the websocket transport.
* Adds support for recursively resolving dnsaddrs (makes go-ipfs compatible with the new bootstrappers).
* Fixes several potential panics/crashes.
* Switches to using pre-defined autorelays instead of trying to find them in the DHT:
  * Avoids selecting random, potentially poor, relays.
  * Avoids spamming the DHT with requests trying to find relays.
  * Reduces the impact of accidentally enabling AutoRelay + RelayHop. I.e., the network won't try to DoS you.
* Modifies the connection manager to not count connections in the grace period towards the connection limit.
  * Pro: New connections don't cause us to close useful, existing connections.
  * Con: Libp2p will keep more connections. Consider reducing your HighWater after applying this patch.
* Improved peer usefulness tracking in bitswap. Frequently used peers will be marked as "important" and the connection manager will avoid closing connections to these peers.
* Includes a new version of the WebUI to fix some issues with the peers map.

## Changelog

- github.com/ipfs/go-ipfs:
  - feat: update the webui to fix some performance issues ([ipfs/go-ipfs#6844](https://github.com/ipfs/go-ipfs/pull/6844))
  - fix: limit SW registration to content root ([ipfs/go-ipfs#6801](https://github.com/ipfs/go-ipfs/pull/6801))
  - fix issue 6760, adding with hash-only, high CPU usage. ([ipfs/go-ipfs#6764](https://github.com/ipfs/go-ipfs/pull/6764))
  - fix(coreapi/add): close the fake repo used when adding with hash-only ([ipfs/go-ipfs#6747](https://github.com/ipfs/go-ipfs/pull/6747))
  - fix bug 6748 ([ipfs/go-ipfs#6754](https://github.com/ipfs/go-ipfs/pull/6754))
  - fix(pin): wait till after fetching to remove direct pin ([ipfs/go-ipfs#6708](https://github.com/ipfs/go-ipfs/pull/6708))
  - pin: fix pin update X Y where X==Y ([ipfs/go-ipfs#6669](https://github.com/ipfs/go-ipfs/pull/6669))
  - namesys: set the correct cache TTL on publish ([ipfs/go-ipfs#6667](https://github.com/ipfs/go-ipfs/pull/6667))
  - build: fix golangci again ([ipfs/go-ipfs#6641](https://github.com/ipfs/go-ipfs/pull/6641))
  - make: move all test deps to a separate module ([ipfs/go-ipfs#6637](https://github.com/ipfs/go-ipfs/pull/6637))
  - fix: close peerstore on stop ([ipfs/go-ipfs#6629](https://github.com/ipfs/go-ipfs/pull/6629))
  - build: fix build when we don't have a full git tree ([ipfs/go-ipfs#6626](https://github.com/ipfs/go-ipfs/pull/6626))
- github.com/ipfs/go-bitswap (v0.0.8-cbb485998356 -> v0.0.8-e37498cf10d6):
  - fix: wait until we finish connecting before we cancel the context ([ipfs/go-bitswap#226](https://github.com/ipfs/go-bitswap/pull/226))
  - engine: tag peers based on usefulness ([ipfs/go-bitswap#191](https://github.com/ipfs/go-bitswap/pull/191))
- github.com/ipfs/go-cid (v0.0.2 -> v0.0.4):
  - fix parsing issues and nits ([ipfs/go-cid#97](https://github.com/ipfs/go-cid/pull/97))
  - Verify that prefix is correct v0 prefix ([ipfs/go-cid#96](https://github.com/ipfs/go-cid/pull/96))
- github.com/multiformats/go-multihash (v0.0.5 -> v0.0.10):
  - Ensure that length of multihash is properly handled ([multiformats/go-multihash#119](https://github.com/multiformats/go-multihash/pull/119))
  - fix murmur3 name  ([multiformats/go-multihash#115](https://github.com/multiformats/go-multihash/pull/115))
  - rename ID to IDENTITY ([multiformats/go-multihash#113](https://github.com/multiformats/go-multihash/pull/113))
 ([multiformats/go-multihash#119](https://github.com/multiformats/go-multihash/pull/119))
- github.com/libp2p/go-flow-metrics (v0.0.1 -> v0.0.3):
  - fix bug in meter traversal logic ([libp2p/go-flow-metrics#11](https://github.com/libp2p/go-flow-metrics/pull/11))
- github.com/libp2p/go-libp2p (v0.0.28 -> v0.0.32):
  - options to configure known relays for autorelay ([libp2p/go-libp2p#705](https://github.com/libp2p/go-libp2p/pull/705))
  - feat(host): recursively resolve addresses ([libp2p/go-libp2p#764](https://github.com/libp2p/go-libp2p/pull/764))
  - mdns: always use interface addresses ([libp2p/go-libp2p#667](https://github.com/libp2p/go-libp2p/pull/667))
- github.com/libp2p/go-libp2p-connmgr (v0.0.6 -> v0.2.1):
  - don't count connections in the grace period against the limit ([libp2p/go-libp2p-connmgr#50](https://github.com/libp2p/go-libp2p-connmgr/pull/50))
- github.com/libp2p/go-libp2p-kad-dht (v0.0.13 -> v0.0.15):
  - metrics: fix memory leak ([libp2p/go-libp2p-kad-dht#390](https://github.com/libp2p/go-libp2p-kad-dht/pull/390))
- github.com/libp2p/go-libp2p-tls (v0.0.1 -> v0.0.2):
  - close the underlying connection when the handshake fails ([libp2p/go-libp2p-tls#39](https://github.com/libp2p/go-libp2p-tls/pull/39))
  - make the error check for not receiving a public key more explicit ([libp2p/go-libp2p-tls#34](https://github.com/libp2p/go-libp2p-tls/pull/34))
  - Fix: Connection Closed after handshake ([libp2p/go-libp2p-tls#37](https://github.com/libp2p/go-libp2p-tls/pull/37))
- github.com/libp2p/go-libp2p-swarm (v0.0.6 -> v0.0.7):
  - fix: don't assume that transports implement stringer ([libp2p/go-libp2p-swarm#134](https://github.com/libp2p/go-libp2p-swarm/pull/134))
- github.com/libp2p/go-ws-transport (v0.0.4 -> v0.0.6):
  - Add mutex for write/close ([libp2p/go-ws-transport#65](https://github.com/libp2p/go-ws-transport/pull/65))

Other:

Update bloom filter libraries to remove unsound usage of the `unsafe` package.

## ✅ Release Checklist

For each RC published in each stage:

- version string in `version.go` has been updated
- tag commit with vX.Y.Z-rcN
- upload to dist.ipfs.io
  1. Build: https://github.com/ipfs/distributions#usage.
  2. Pin the resulting release.
  3. Make a PR against ipfs/distributions with the updated versions, including the new hash in the PR comment.
  4. Ask the infra team to update the DNSLink record for dist.ipfs.io to point to the new distribution.
- cut a pre-release on [github](https://github.com/ipfs/go-ipfs/releases) and upload the result of the ipfs/distributions build in the previous step.

Checklist:

- [x] **Stage 0 - Automated Testing**
  - [x] Feature freeze. If any "non-trivial" changes (see the footnotes of [docs/releases.md](https://github.com/ipfs/go-ipfs/tree/master/docs/releases.md) for a definition) get added to the release, uncheck all the checkboxes and return to this stage.
  - [x] Automated Testing (already tested in CI) - Ensure that all tests are passing, this includes:
    - [x] unit, sharness, cross-build, etc (`make test`)
    - [x] lint (`make test_go_lint`)
    - [x] [interop](https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs)
    - [x] [go-ipfs-api](https://github.com/ipfs/go-ipfs-api)
    - [x] [go-ipfs-http-client](https://github.com/ipfs/go-ipfs-http-client)
    - [x] [WebUI](https://github.com/ipfs-shipyard/ipfs-webui)
- [x] **Stage 1 - Internal Testing**
  - [x] CHANGELOG.md has been updated
    - use [`./bin/mkreleaselog`](https://github.com/ipfs/go-ipfs/tree/master/bin/mkreleaselog) to generate a nice starter list
  - [x] Infrastructure Testing:
    - [x] Deploy new version to a subset of Bootstrappers
       - Deploying to mars.
    - [x] Deploy new version to a subset of Gateways
    - ~Deploy new version to a subset of Preload nodes~
    - ~Collect metrics every day. Work with the Infrastructure team to learn of any hiccup~
  - [x] IPFS Application Testing -  Run the tests of the following applications:
    - [x] [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop) - @hacdias
    - [x] [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion) - @lidel
    - ~[NPM on IPFS](https://github.com/ipfs-shipyard/npm-on-ipfs)~
- ~**Stage 2 - Community Dev Testing**~
  - *skipping because this is a patch release*
- [x] **Stage 3 - Community Prod Testing**
  - [x] Documentation
    - [x] Ensure that [CHANGELOG.md](https://github.com/ipfs/go-ipfs/tree/master/CHANGELOG.md) is up to date
    - [x] Ensure that [README.md](https://github.com/ipfs/go-ipfs/tree/master/README.md)  is up to date
    - [x] Ensure that all the examples we have produced for go-ipfs run without problems
    - [x] Update HTTP-API Documentation on the Website using https://github.com/ipfs/http-api-docs
  - [x] Invite the IPFS [_early testers_](https://github.com/ipfs/go-ipfs/tree/master/docs/EARLY_TESTERS.md) to deploy the release to part of their production infrastructure.
  - [x] Invite the wider community through (link to the release issue):
    - [x] [discuss.ipfs.io](https://discuss.ipfs.io/c/announcements)
    - [ ] Twitter
    - [x] IRC
- [x] **Stage 4 - Release**
  - [x] Final preparation
    - [x] Verify that version string in [`repo/version.go`](https://github.com/ipfs/go-ipfs/tree/master/repo/version.go) has been updated
    - [x] tag commit with vX.Y.Z
    - [x] update release branch to point to release commit (`git merge vX.Y.Z`).
    - [x] Release published
      - [x] to [dist.ipfs.io](https://dist.ipfs.io)
      - [x] to [npm-go-ipfs](https://github.com/ipfs/npm-go-ipfs)
      - [x] to [chocolatey](https://chocolatey.org/packages/ipfs)
      - [x] to [github](https://github.com/ipfs/go-ipfs/releases)
  - [x] Publish a Release Blog post (at minimum, a c&p of this release issue with all the highlights, API changes, link to changelog and thank yous)
  - [x] Broadcasting (link to blog post)
    - [x] Twitter
    - [x] IRC
    - [x] [Reddit](https://reddit.com/r/ipfs)
    - [x] [discuss.ipfs.io](https://discuss.ipfs.io/c/announcements)
    - [x] Announce it on the [IPFS Users Mailing List](https://groups.google.com/forum/#!forum/ipfs-users)
- [x] **Post-Release**
  - [x] Bump the version in `version.go` to `vX.(Y+1).0-dev`.
  - [x] Create an issue using this release issue template for the _next_ release.
  - [x] Make sure any last-minute changelog updates from the blog post make it back into the CHANGELOG.

## ❤️ Contributors

| Contributor | Commits | Lines ± | Files Changed |
|-------------|---------|---------|---------------|
| Steven Allen | 52 | +1866/-578 | 102 |
| vyzo | 12 | +167/-90 | 22 |
| whyrusleeping | 5 | +136/-52 | 7 |
| Roman Proskuryakov | 7 | +94/-7 | 10 |
| Jakub Sztandera | 3 | +58/-13 | 7 |
| hucg | 2 | +31/-11 | 2 |
| Raúl Kripalani | 2 | +7/-33 | 6 |
| Marten Seemann | 3 | +27/-10 | 5 |
| Marcin Rataj | 2 | +26/-0 | 5 |
| b5 | 1 | +2/-22 | 1 |
| Hector Sanjuan | 1 | +11/-0 | 1 |
| Yusef Napora | 1 | +4/-0 | 1 |

Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:

- Check the issues with the `help wanted` label in the [go-ipfs repo](https://github.com/ipfs/go-ipfs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
- Join the discussion at [discuss.ipfs.io](https://discuss.ipfs.io/) and help users finding their answers.
- Join the [🚀 IPFS Core Implementations Weekly Sync 🛰](https://github.com/ipfs/team-mgmt/issues/992) and be part of the action!

## ⁉️ Do you have questions?

The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](http://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode, which is also [accessible through our Matrix bridge](https://riot.im/app/#/room/#freenode_#ipfs:matrix.org).

**Comment by Stebalien** [comment]
@alanshaw could I borrow your expertise at writing wonderful, engaging release notes for this release?

(I understand that might be a bit tricky given that you weren't directly involved in most of the changes; we can discuss this on Monday and/or I can try writing a soulless skeleton).

**Comment by Stebalien** [comment]
RC1 at [/ipfs/QmRxq967po6BcGW68wbUvwin758dkRiejAKmyHDm9wqaRB/go-ipfs/v0.4.23-rc1/](https://ipfs.io/ipfs/QmRxq967po6BcGW68wbUvwin758dkRiejAKmyHDm9wqaRB/go-ipfs/v0.4.23-rc1/). I'll update dist.ipfs.io later.

**Comment by alanshaw** [comment]
> @alanshaw could I borrow your expertise at writing wonderful, engaging release notes for this release?

I'll give it a go!

**Comment by lidel** [comment]
@Stebalien is it too late to include WebUI fixes from https://github.com/ipfs/go-ipfs/pull/6778#issuecomment-561595975?
Commit to cherrypick is 5ab7a70131cb69f2999e86a24a82dc451c0aa0a0

**Comment by Stebalien** [comment]
No, we can include that. Could you send a patch against the current RC branch to use the new version?

**Comment by Stebalien** [comment]
I'm releasing an RC2 with two changes:

1. The backported WebUI update. This should probably count as a "feature" but it's not likely to cause wide-scale regressions and it includes a bunch of important bug fixes.
2. A patch to fix a memory leak from the DHT that I intended to include but didn't. Luckily, we have our handy release checklist and this was caught when profiling a bootstrapper.

**Comment by Stebalien** [comment]
Early testers: We're entering stage 3 of this patch release. As with the previous release, this release only includes cherry-picked bug fixes (quite a few of them) so we don't expect any regressions or significant performance changes. However, we're still inviting you to do a quick sanity check before the final release.

Note: We have cherry-picked a few patches that could cause interesting behavioral changes:

If you're relying on AutoRelay, we're no longer picking random relays from the DHT. Instead, we've hard-coded a list of known-good relays. This is a stop-gap to deal with the fact that many of the public relays were unreliable, causing nodes using AutoRelay to continuously crawl the network looking for a good relay.

The connection manager no longer counts connections in the "grace period" (default is 30s) against the limit. Before this change, nodes with many inbound connections would disconnect from _all_ long-term peers as connections in the grace period would take the entire quota. You may need to adjust your connection limits to account for this change.

Finally, bitswap now tracks which peers we frequently exchange data with and asks the connection manager to avoid closing connections to these peers.

- [ ] Infura (@MichaelMure)
- [ ] Textile (@sanderpick)
- [ ] Pinata (@obo20)
- [ ] RTrade (@postables)
- [ ] QRI (@b5)
- [ ] Siderus (@koalalorenzo)

**Comment by MichaelMure** [comment]
Thanks for the release :)

Just a note: consider pinging the early testers when you make the release (aka 5 days ago) instead of at stage 3 so we can do the testing with a bit more room.

**Comment by Stebalien** [comment]
@MichaelMure you're right, I should have pinged the early testers when creating the release issue. I'll add that to the release process.

**Comment by MichaelMure** [comment]
What about fixing https://github.com/ipfs/go-ipfs/issues/6795 as well ?

**Comment by Stebalien** [comment]
`go get github.com/ipfs/go-ipfs/cmd/ipfs@v0.4.23-rc2` works for me.

**Comment by bonedaddy** [comment]
I'm currently testing rc2 in dev. I wont be able to switch over any production nodes until friday during maintenance however.

**Comment by Stebalien** [comment]
@bonedaddy SGTM. We're planning on cutting a release either late friday or early next monday but I'm willing to wait to get this right.

**Comment by alanshaw** [comment]
I'm on this now - sorry for the delay. Here's my intro:

> Yes, that's right the next feature release of go-ipfs (0.5.0) is, well, running a tiny bit behind schedule. In the mean time though we have patches, and I'm not talking pirate eye patches, I'm talking bug fixes. We're hunting these bugs like they're pokemon, and jeez, do we come across some rare and difficult to fix ones? - you betcha.

**Comment by b5** [comment]
@alanshaw playing to both pirate and pokemon crowds in a single paragraph? Deep.

I'll upgrade our deps & run tests ASAP & try to report back before the end of the week. @Stebalien I'm hoping you're still a few timezones behind me 😄

**Comment by RubenKelevra** [comment]
> Update bloom filter libraries to remove unsafe usage of unsafe.

You're sure this sentence should be that way? 🤔

**Comment by Stebalien** [comment]
I guess "unsound" usage of unsafe is more accurate.

**Comment by RubenKelevra** [comment]
> I guess "unsound" usage of unsafe is more accurate.

I thought you accidentally wrote 'unsafe' the second time, but actually meant something else :)

**Comment by rklaehn** [comment]
> we're strongly considering switching to TLS by default in go-ipfs 0.5.0 and dropping SECIO support. 

I hope there will be a long grace period where both TLS and SECIO are supported. We got a lot of ipfs nodes in production, and given the trouble we had upgrading last time we are not going to immediately upgrade all nodes.

Also, it is very important for us that interop between rust-libp2p and js-ipfs via TLS is confirmed to work before disabling SECIO.

**Comment by RubenKelevra** [comment]
I've read that as:

They lean strongly towards setting the TLS as first option on a connection. Which means a handshake with SECIO is the second option, when TLS doesn't work.

This means new connections with SECIO will take longer.

The idea is to deprecate SECIO, so any new implementation should only implement TLS, any existing peer up (including 0.5) will support both and keeps the network running.

**Comment by Stebalien** [comment]
> I hope there will be a long grace period where both TLS and SECIO are supported.

TLS has been supported for quite some time (since go-ipfs 0.4.21, 30 May 2019). The issue fixed in this release was just a bug that could cause the handshake to fail in some rare cases.

The current plan is to _drop_ SECIO support in go-ipfs 0.5.0, assuming we have full interoperability with js-ipfs and rust-libp2p.

Note: go-ipfs 0.5.0 will not work with go-ipfs nodes less than 0.4.21 anyways as it introduces a feature that triggers a bug in 0.4.20 (and before).

> We got a lot of ipfs nodes in production, and given the trouble we had upgrading last time we are not going to immediately upgrade all nodes.

0.4.23 is a patch release on 0.4.22 which is a patch release on 0.4.21. If you're running 0.4.22 or 0.4.21, the upgrade should be very smooth.

> Also, it is very important for us that interop between rust-libp2p and js-ipfs via TLS is confirmed to work before disabling SECIO.

The plan is to introduce a Noise transport in all languages as it's significantly easier to implement correctly. There are ongoing efforts in rust, js, and go.

**Comment by alanshaw** [comment]
Still working on this but here's what I got so far (I'll update this post as more is written):

---

# go-ipfs 0.4.23

> Would sir/madam care for another patch release while they wait?

Yes, that's right the next feature release of go-ipfs (0.5.0) is, well, running a tiny bit behind schedule. In the mean time though we have patches, and I'm not talking pirate eye patches, I'm talking bug fixes. We're hunting these bugs like they're Pokemon, and jeez, do we come across some rare and difficult to fix ones? - you betcha.

Alright, enough funny business, what's the deal? Ok so, I don't want to _alarm_ anyone but this release has some **critical fixes** and if you're using go-ipfs or know someone who is then you and your friends need to slide into your upgrade pants and give those IPFS nodes a good wipe down ASAP.

## 🤝 Fixed Spontaneous TLS Disconnects

If this isn't reason enough to upgrade I don't know what is. Turns out, a TLS handshake _may_ have accidentially been unintentionally aborted for no good reason 😱. I shouldn't need to tell you this but \*wispers\* that's really bad for connectivity! Excellent news though, it's fixed, and if you want to stay well connected in the glorious future of p2p communications then you should definitely upgrade.

## 😱 Fixed Panics and Crashes

Panicing won't help, in life, and also in golang. Stay calm and breathe slowly. We patched a number of panics and crashes that were uncovered, including a panic seen commonly in the websocket transport. High ten 🙌?

## 🔁 Fixed Resursive Resolving of dnsaddr Multiaddrs

Imagine this: rolling out a brand spanking new set of bootstrap nodes only to discover their new addresses are not resolvable. I know right - bad news bears 🐻!?

Ok, "bear" this in mind: you want to keep all your DNS TXT records [below 512 bytes to avoid UDP fragmentation](https://serverfault.com/questions/840241/do-dns-queries-always-travel-over-udp), otherwise you'll get a truncated reply and have to connect with TCP to get all the records. If you have lots of dnsaddr TXT records then it can be more efficient to use recursive resolving than to get a truncated reply and go through the famous 18-way SYN, SYN-ACK ACK, ACK-SYN, ACK-ACK (...etc, etc) TCP handshake, not to mention the fact that go-ipfs will not even try to fallback to TCP 😅.

Anyway, long story short. We fixed that. You're welcome.

## 📻 Connection Manager Tuned Correctly

The Connection Manager has been tuned to better prioritise existing connections by not counting new connections in the "grace" period towards connection limits. New connections are like new friends. You can't hang out with everyone all the time, I mean, it just gets difficult to book a resturant after a while.

You also wouldn't stop being friends with Jane just because you met Sarah _once_ on the train. You and Jane have history, think of everything you've been through. Remember that time when Jane's dog, Dave, ran away? I know, it's a weird name for a dog, I mean who gives a human name to a dog anyway, but I guess that's one of the reasons you like Jane. Anyway, she lost her dog and you both looked all around town for it, you were about to give up but then you heared faint wimpering as you were walking back to the house. Dave had somehow managed to fall into the old abandoned well!

You see?! History! ...and, erh, what was I saying? Oh yeah, Connection Manager - new connections don't cause us to close useful, existing connections (like Jane), but it does mean you'll keep more connections in total. Maybe consider reducing the `HighWater` setting in your config.

**More fixes:**

* Fixes build on go 1.13
* Fixes an issue where we may not connect to providers in bitswap.
* Switches to using pre-defined autorelays instead of trying to find them in the DHT:
  * Avoids selecting random, potentially poor, relays.
  * Avoids spamming the DHT with requests trying to find relays.
  * Reduces the impact of accidentally enabling AutoRelay + RelayHop. I.e., the network won't try to DoS you.
* Improved peer usefulness tracking in bitswap. Frequently used peers will be marked as "important" and the connection manager will avoid closing connections to these peers.
* Includes a new version of the WebUI to fix some issues with the peers map.

**Comment by Stebalien** [comment]
I'm planning on cutting the release by EOD today (PST) or tomorrow morning at the latest.

Does anyone have any regressions to report?

**Comment by bonedaddy** [comment]
Maintenance was postponed so I didn't get a chance to deploy one node in production, but the node in dev seems fine.

Output from `pmap` seems to be slightly improved with `922744152K` vs the usual `999xxxxxxxK`

**Comment by Stebalien** [comment]
> Output from pmap seems to be slightly improved with 922744152K vs the usual 999xxxxxxxK

Good to hear. We fixed at least one memory leak.

**Comment by Stebalien** [comment]
I'm going to include one more last minute fix in this release: https://github.com/ipfs/go-ipfs/issues/6418 (not sure how I missed this). The fix is trivial (https://github.com/libp2p/go-libp2p-swarm/commit/50cbe2b6652f8042358a9d52bf6a14553698c8db) so I'm not going to bother with another RC.

**Comment by b5** [comment]
LGTM

**Comment by Stebalien** [comment]
The release is up and announced. Thanks everyone!
